api/repositories/pod_repository_test.go

```diff
@@ -0,0 +1,84 @@
+package repositories_test
+
+import (
+	"code.cloudfoundry.org/korifi/api/repositories"
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PodRepository", func() {
+	var (
+		podRepo *repositories.PodRepo
+		org     *korifiv1alpha1.CFOrg
+		space   *korifiv1alpha1.CFSpace
+		pod     *corev1.Pod
+		appGUID string
+		app     *repositories.AppRecord
+	)
+	BeforeEach(func() {
+
+		podRepo = repositories.NewPodRepo(userClientFactory)
+		org = createOrgWithCleanup(ctx, prefixedGUID("org"))
+		space = createSpaceWithCleanup(ctx, org.Name, prefixedGUID("space"))
+
+		appGUID = uuid.NewString()
+		app = &repositories.AppRecord{
+			GUID:      appGUID,
+			Revision:  "1",
+			SpaceGUID: space.Name,
+		}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "podname-2",
+				Namespace: space.Name,
+				Labels: map[string]string{
+					"korifi.cloudfoundry.org/app-guid":     appGUID,
+					"korifi.cloudfoundry.org/version":      "1",
+					"korifi.cloudfoundry.org/process-type": "web",
+				}},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "web",
+						Image: "nginx",
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, pod)).To(Succeed())
+
+	})
+	Describe("DeletePod", func() {
+		When("the user is not a a SpaceDeveloper", func() {
```

Comment:
This `When` can be nested in the describe as it could be looked at as the default case